lib/utils/search.js

```diff
@@ -32,12 +38,47 @@ const getWeight = (searchTerm, words, resource, resourceIndex) => {
   return weight;
 };
 
+const weightByCategories = (ids, items) => {
+  return JSON.parse(JSON.stringify(items)).map(item => {
+    item.resources = item.resources.map(resource => {
+      resource.weight = getCategoryWeight(ids, items, resource);
+      return resource;
+    });
+    return item;
+  });
+};
+
+const getCategoryWeight = (ids, categorisedResources, resource) => {
+  const categoriesResourceIsIn = categorisedResources
+    .filter(category => {
+      return category.resources.some(r => r.id == resource.id);
+    })
+    .map(x => x.id);
```

Comment:
Oh, so with the data we get back, we get a single resource duplicated over every service it is in?
Do I understand this correctly? So this function gets the Ids of all of the categories that the input resource is in?

Reply:
Yes it seems like an odd structure to me as we have an object that makes it to the front-end and isn't necessarily transformed into the appropriate structure until quite late. I consider the 'flattenSearchResults' method tech debt.

Reply:
When you say every service it is in I assume you mean every category it's in*